armi/reactor/blocks.py

```diff
@@ -1732,6 +1732,35 @@ def getUraniumMassEnrich(self):
         u8 = self.getMass("U238")
         return u5 / (u8 + u5)
 
+    def getInputHeight(self) -> float:
+        """Determine the input height from blueprints.
+
+        Returns
+        -------
+        float
+            Height for this block pulled from the blueprints.
+
+        Raises
+        ------
+        AttributeError
+            If no ancestor of this block contains the input blueprints. Blueprints are
+            usually stored on the reactor object, which is typically an ancestor of
+            the block (block -> assembly -> core -> reactor). However, this may be the case
+            when creating blocks from scratch in testing where the entire composite
+            tree may not exist.
+        """
+        ancestorWithBp = self.getAncestor(
+            lambda o: getattr(o, "blueprints", None) is not None
+        )
+        if ancestorWithBp is not None:
+            parent: composites.Composite = self.parent
```

Comment:
seems like we could leave this line out. `self.parent` is only called twice.

#not-blocking